Receipt: Add payment proof

dennisreimann opened the pull request:
We now display a payment proff on the receipt page:

- Onchain: Transaction Id
- Lightning: Preimage

Closes #4685.

![grafik](https://user-images.githubusercontent.com/886/225935082-d9fe8c04-b76e-4320-a4d7-72e56cee63b6.png)

dennisreimann commented:
TODO:

- [ ] Onchain: Remove duplicate Payment Proof
- [ ] Offchain: Remove Transaction Id

dennisreimann commented:
Improved the view and added a view component for displaying truncated text — it's configurable, displays things properly when printed and optionally shows a copy button. We can re-use it elsewhere once this is merged.

![grafik](https://user-images.githubusercontent.com/886/227369382-0c1d637d-b92a-4067-b852-e63d8c36c745.png)

dstrukt commented:
Only thought w.r.t to the TX id's ( i ) is that it might not be immediately obvious it's a click away link .. wondering if we'll want to use a more common icon for that .. thoughts?

Otherwise, looks much better and cleaner .. going to explore some more options tomorrow when I'm in Figma.

dennisreimann commented:
I reused what we have on the labels, but am willing to adapt both if you find a better solution :)

NicolasDorier commented:
Something is strange.

The `Transaction Id` in on-chain term is only the tx hash, here it is the outpoint. (`txid-idx`)
The `Proof` should be the outpoint.

Here they are the same.

NicolasDorier commented on BTCPayServer/Components/TruncateCenter/TruncateCenter.cs:
```diff
@@ -0,0 +1,21 @@
+
+namespace BTCPayServer.Components.TruncateCenter
+{
+    public class TruncateCenter : ViewComponent
```
small comment to understand what is that?

dennisreimann replied:
Added in [83eadc2](https://github.com/btcpayserver/btcpayserver/pull/4782/commits/83eadc2bcb9b6014f9f073c6376e7a3d765b09a3).

NicolasDorier commented on BTCPayServer/Services/Invoices/InvoiceEntity.cs:
```diff
@@ -1330,5 +1330,6 @@ public interface CryptoPaymentData
 
         PaymentType GetPaymentType();
         string GetDestination();
+        string GetPaymentProof();
```
Not sure it's good idea to make this interface bigger. If anything, we should probably break it down later.
Anyway, leave it like that for now, I may change your PR later.

dennisreimann commented:
@NicolasDorier I've now removed the Transaction ID, as we use `GetPaymentId()` in several places and I didn't want to touch it, because it most likely assumed the outpoint.

Also this let's us display everything consistent and independent of the payment method. Added the link to the payment proof, so that it is also still present. 

![grafik](https://user-images.githubusercontent.com/886/227779680-805c2366-70cc-4872-a97b-bfb1a3f7dcc2.png)

You can take over now :)

NicolasDorier commented:
I removed interface and just hard coded for LN and bitcoin payments.

This is less flexible, but we will soon refactor all of this anyway.
